statsmodels/base/model.py

```diff
@@ -510,6 +510,149 @@ def hess(params, *args):
+            transformed.
+        k_params : int or None
+            If None, then we try to infer from start_params or model.
+        **fit_kwds : keyword arguments
+            fit_kwds are used in the optimization of the transformed model.
+
+        Returns
+        -------
+        results : Results instance
+
+        """
+        # wee need to append index of extra params to keep_index as in NegativeBinomial
+        if hasattr(self, 'k_extra') and self.k_extra > 0:
+            # we cannot change the original, TODO: should we add keep_index_params?
+            keep_index = np.array(keep_index, copy=True)
+            keep_index_p = np.concatenate((keep_index,
+                                           list(range(self.exog.shape[1] + self.k_extra))[-self.k_extra:]))
+        else:
+            keep_index_p = keep_index
+
+        # not all models support start_params, drop if None, hide them in fit_kwds
+        if start_params is not None:
+            fit_kwds['start_params'] = start_params[keep_index_p]
+            k_params = len(start_params) # ignore k_params in this case, or verify consisteny?
+
+        # build auxiliary model and fit
+        init_kwds = self._get_init_kwds()
+        mod_constr = self.__class__(self.endog, self.exog[:, keep_index], **init_kwds)
+        res_constr = mod_constr.fit(**fit_kwds)
+        #switch name, only need keep_index for params belos
+        keep_index = keep_index_p
+
+        if k_params is None:
+            k_params = self.exog.shape[1]
+            k_params += getattr(self, 'k_extra', 0)
+
+
+        params_full = np.zeros(k_params)
+        params_full[keep_index] = res_constr.params
+
+
+        # create dummy results Instance, TODO: wire up properly
+        # TODO: this could be moved into separate private method if needed
+        # discrete L1 fit_regularized doens't reestimate AFAICS
+        #res = self.fit(maxiter=0, method='nm', disp=0,
+        # RLM doesn't have method, disp nor warn_convergence keywords
+        # OLS, WLS swallows extra kwds with **kwargs, but doesn't have method='nm'
+        try:
+            # Note: addding full_output=False causes currently exceptions
+            res = self.fit(maxiter=0, disp=0, method='nm', skip_hessian=True,
+                           warn_convergence=False, start_params=params_full)
+            # we get a wrapper back
+        except (TypeError, ValueError):
+            res = self.fit()
+
+        # Warning: make sure we are not just changing the wrapper instead of results #2400
+        if hasattr(res_constr.model, 'scale'):
+            # GLM problem, see #2399
+            res.model.scale = res._results.scale = res_constr.model.scale
+
+        if hasattr(res_constr, 'mle_retvals'):
+            res.mle_retvals = res_constr.mle_retvals
+            # not available for not scipy optimization, e.g. glm irls
+#             res.mle_retvals['fcall'] = res_constr.mle_retvals.get('fcall', np.nan)
```

Comment:
pls delete commented-out

Reply:
Either delete or document.  In its current form, this commented-out code is worse-than-useless for a 3rd-party reader.  Experience suggests this is all too likely to retain its current form years from now.